lib/opencensus/trace/span_context.rb

```diff
@@ -51,26 +62,41 @@ class << self
         #
         # @param [String] header A Trace-Context header (optional)
         # @param [Hash] rack_env The Rack environment hash (optional)
+        # @param [#serialize,#rack_header_name] formatter The formatter to use
+        #     when propagating span context. Optional: If omitted, uses the
+        #     TraceContext formatter.
         #
         # @return [SpanContext]
         #
         def create_root header: nil, rack_env: nil, formatter: nil
-          formatter ||= Formatters::DEFAULT
-          header ||= rack_env["HTTP_TRACE_CONTEXT"] if rack_env
+          detected_formatter = detect_formatter rack_env
```

Comment:
I'm confused about the difference between formatter and detected_formatter. It looks like they could be different, but what does that mean, and when do we use one vs the other?

Reply:
We're currently deciding which formatter to use from the available incoming header names. If you provide the header directly, we can't use that strategy.

It's possible that we could try to parse the directly provided header against the different context formats rather than based on the header names

Reply:
Actually, implicitly depending on the detected request formatter seems wrong. If someone sends a Trace-Context header and you're using CloudTrace internally, the faraday middleware might send along the Trace-Context header instead. The developer should configure the middleware to use the formatter their system is using.